ui/js/src/App.tsx

```diff
@@ -122,56 +52,9 @@ export const AppLayout: React.FC<LayoutProps> = function (props: LayoutProps) {
       <ErrorBar
         key={notificationText}
         text={notificationText}
-        dismissNotification={dismissNotification}
       />
     </View>
   );
 };
 
-type LayoutProps = {
-  activeWorkContext: string | undefined;
-  addNotification: (text: string) => void;
-  completeAuthentication: (token: string) => void;
-  createTodo: (description: string) => void;
-  dismissNotification: () => void;
-  filteredTodos: Todo[];
-  isLoading: boolean;
-  labels: Label[];
-  moveTodo: (operation: MoveTodoOperation) => void;
-  notificationQueue: string[];
-  selectedPickerLabels: { [label: string]: boolean };
-  setEditTodoId: (id: number | null) => void;
-  setLabelTodoId: (id: number | null) => void;
-  setWorkContext: (workContext: string) => void;
-  toggleLabel: (label: string) => void;
-  toggleShowCompletedTodos: () => void;
-  toggleShowLabelFilter: () => void;
-  updateTodo: (todoPatch: TodoPatch) => void;
-  updateTodoLabels: (labels: string[]) => void;
-  workspace: WorkspaceState;
-};
-
-const mapStateToProps = (state: RootState) => {
-  return {
-    labels: state.labelsApi.entries,
-    workspace: state.workspace,
-    notificationQueue: state.notifications.notificationQueue,
-  };
-};
-const mapDispatchToProps = {
-  addNotification,
-  completeAuthentication,
-  createTodo,
-  dismissNotification,
-  moveTodo,
-  setEditTodoId,
-  setLabelTodoId,
-  setWorkContext,
-  toggleLabel,
-  toggleShowCompletedTodos,
-  toggleShowLabelFilter,
-  updateTodo,
-  updateTodoLabels,
-};
-const connector = connect(mapStateToProps, mapDispatchToProps);
```

Comment:
I like how using the hooks simplifies all this, and avoids having to pass down these props through the hierarchy.

Although, I was curious about using connect at the root level (before these changes) instead of letting each component taking care of mapping the required fields and actions. Was it to avoid writing the connect logic multiple times?

Reply:
I'm not certain.  I think it was a best practice aiming to have a central connected "controller" component and then individual controlled or UI specific components.  I know that made it easier to use tools like Storybook, etc.

I still haven't played w/ the new setup enough, but I suspect I may end up adding back passing data via props in some cases to make components easier to configure in storybook.  But for now it's also been very easy to bind storybook to redux w/ an initial state.

I also haven't worked in an app w/ multiple connect setups so I'm not sure if there are any downsides of limitations.